### PR TITLE
Added docs to the interrupts crate

### PR DESCRIPTION
**Pull request by nabeelomer**
This should fix https://github.com/intermezzOS/kernel/issues/77.
This is my first time writing docs from scratch, please let me know what you think.

### CODE REVIEW

File: interrupts/src/lib.rs [hunk 1]
```diff
@@ -1,3 +1,7 @@
+//! Interrupts
```
Comment (steveklabnik) [inline interrupts/src/lib.rs]: Putting the name here isn't important; The docs will already say this is for the interrupts crate

File: interrupts/src/lib.rs [hunk 2]
```diff
@@ -75,8 +82,14 @@ macro_rules! make_idt_entry {
     }};
 }
 
+/// The Interrupt Descriptor Table
+///
+/// The CPU will look at this table to find the appropriate interrupt handler.
+///
```
Comment (steveklabnik) [inline interrupts/src/lib.rs]: this extra line should be removed

File: interrupts/src/lib.rs [hunk 3]
```diff
@@ -75,8 +82,14 @@ macro_rules! make_idt_entry {
...
 static IDT: Mutex<[IdtEntry; 256]> = Mutex::new([IdtEntry::MISSING; 256]);
 
+/// Pointer to the Interrupt Descriptor Table
+///
```
Comment (steveklabnik) [inline interrupts/src/lib.rs]: this line should be removed

File: interrupts/src/lib.rs [hunk 4]
```diff
@@ -85,6 +98,8 @@ pub struct IdtRef {
 unsafe impl Sync for IdtRef {}
 
 impl IdtRef {
+    /// Creates a new pointer struct to the IDT.
+    ///
```
Comment (steveklabnik) [inline interrupts/src/lib.rs]: This line should be removed.

File: interrupts/src/lib.rs [hunk 5]
```diff
@@ -93,17 +108,19 @@ impl IdtRef {
...
-        unsafe {
-            dtables::lidt(&r.ptr)
-        };
+        unsafe { dtables::lidt(&r.ptr) };
 
         r
     }
 
+    /// Sets an IdtEntry as a handler for interrupt specified by `index`
+    ///
```
Comment (steveklabnik) [inline interrupts/src/lib.rs]: This line should be removed

File: interrupts/src/lib.rs [hunk 6]
```diff
@@ -93,17 +108,19 @@ impl IdtRef {
...
         // IDT.
-        unsafe {
-            dtables::lidt(&r.ptr)
-        };
+        unsafe { dtables::lidt(&r.ptr) };
 
         r
     }
 
+    /// Sets an IdtEntry as a handler for interrupt specified by `index`
```
Comment (steveklabnik) [inline interrupts/src/lib.rs]: Need a period at the end of this sentence.

File: interrupts/src/lib.rs [hunk 7]
```diff
@@ -93,17 +108,19 @@ impl IdtRef {
...
+    ///
     pub fn set_handler(&self, index: usize, entry: IdtEntry) {
         self.idt.lock()[index] = entry;
     }
 
+    /// Enables interrupts
```
Comment (steveklabnik) [inline interrupts/src/lib.rs]: need a period at the end of this sentence

File: interrupts/src/lib.rs [hunk 8]
```diff
@@ -93,17 +108,19 @@ impl IdtRef {
...
+    ///
```
Comment (steveklabnik) [inline interrupts/src/lib.rs]: this line should be removed

**Comment by nabeelomer** [comment]
@steveklabnik I fixed the problems that you pointed out.

**Comment by steveklabnik** [comment]
Thank you so much!

**Pull request merged**